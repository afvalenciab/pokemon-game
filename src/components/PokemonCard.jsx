import React,{ useContext } from 'react';
import { PlayerContext } from '../context/context';
import '../assets/styles/CarouselItem.scss';

const PokemonCard = (props) => {
  const { name, urlPicture } = props;
  const player = useContext(PlayerContext);

  const handleClickPokemon = () => {
    
    console.log(`El ${player} eligio el pokemon ${name}`);
  };

  return (
    <div className='player--carousel-item' onClick={handleClickPokemon}>
      <figure>
        <img src={urlPicture} alt={name} />
      </figure>
      <p>{name}</p>
    </div>
  );
};

export default PokemonCard;
