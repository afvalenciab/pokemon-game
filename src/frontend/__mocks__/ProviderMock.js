import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import initialState from './initialState';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)));
const history = createBrowserHistory();

const ProviderMock = (props) => {
  return(
    <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
    </Provider>
  );
};

export default ProviderMock;