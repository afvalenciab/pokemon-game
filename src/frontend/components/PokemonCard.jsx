import React,{ useContext } from 'react';
import { connect } from 'react-redux';
import { setPokemonPlayer, setLoadingPokemon } from '../actions';
import { PlayerContext } from '../context/context';
import getPokemonData from '../utils/getPokemonData';
import '../assets/styles/CarouselItem.scss';

const PokemonCard = (props) => {
  const { name, urlPicture, setPokemonPlayer, setLoadingPokemon } = props;
  const player = useContext(PlayerContext);

  const handleClickPokemon = async () => {
    setLoadingPokemon({ player, value: true});
    const pokemonData = await getPokemonData(name, player);
    setPokemonPlayer({pokemonData, player});
  };

  return (
    <div className='player--carousel-item' onClick={handleClickPokemon}>
      <figure>
        <img src={urlPicture} alt={name} />
      </figure>
      <p>{name}</p>
    </div>
  );
};

const mapDispatchToProps = {
  setPokemonPlayer,
  setLoadingPokemon,
};

export default connect(null, mapDispatchToProps)(PokemonCard);
