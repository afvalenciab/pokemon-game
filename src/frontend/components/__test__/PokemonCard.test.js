import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import PokemonCard from '../PokemonCard';

describe('PokemonCard testing', () => {
  test('Match Snapshot', () => {
    const pokemonCard = create(
      <ProviderMock>
        <PokemonCard />
      </ProviderMock>
    );
    expect(pokemonCard.toJSON()).toMatchSnapshot();
  });
});