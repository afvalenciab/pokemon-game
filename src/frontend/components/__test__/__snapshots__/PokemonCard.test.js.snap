// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard testing Match Snapshot 1`] = `
<div
  className="player--carousel-item"
  onClick={[Function]}
>
  <figure>
    <img />
  </figure>
  <p />
</div>
`;
